src/main/java/com/google/cloud/discotoproto3converter/proto3/Message.java

```diff
@@ -38,6 +59,49 @@ public Message(String name, boolean ref, boolean isEnum, String description) {
     return fields;
   }
 
+  public Map<Integer, Field> getFieldsWithNumbers() {
+    Map<Integer, Field> fieldsWithNumbers = new LinkedHashMap<>();
+
+    List<Field> sortedFields =
+        fields.stream()
+            .skip(isEnum ? 1 : 0)
+            .sorted(Comparator.comparing(Field::getName))
+            .collect(Collectors.toList());
+    if (isEnum) {
+      sortedFields.add(0, fields.get(0));
+    }
+
+    for (Field f : sortedFields) {
+      if (fieldsWithNumbers.isEmpty() && isEnum()) {
+        fieldsWithNumbers.put(0, f);
+        continue;
+      }
+      int fieldNumber = getFieldNumber(f.getName());
+      while (fieldsWithNumbers.containsKey(fieldNumber)) {
+        fieldNumber = incrementFieldNumber(fieldNumber);
+      }
+      fieldsWithNumbers.put(fieldNumber, f);
+    }
+
+    return fieldsWithNumbers;
+  }
+
+  private int getFieldNumber(String fieldName) {
+    int fieldNumber = (fieldName.hashCode() << 3) >>> 3;
+    if (fieldNumber == 0 || (fieldNumber >= 19000 && fieldNumber <= 19999)) {
+      fieldNumber = 20000 + ((fieldNumber % 19000) + 1) * 536314;
+    }
+    return fieldNumber;
+  }
+
+  private int incrementFieldNumber(int fieldNumber) {
+    int incrementedFieldNumber = fieldNumber + 1;
+    if ((fieldNumber >= 19000 && fieldNumber <= 19999)) {
+      incrementedFieldNumber = 20000;
+    }
+    return incrementedFieldNumber;
```

Comment:
Nice catch! Yeah, I doubt it would ever happen, but it is good to have it rigorous especially given that it is super easy to implement. Added the logic.